src/mmw/apps/geoprocessing_api/calcs.py

```diff
@@ -135,8 +135,33 @@ def catchment_water_quality(geojson):
             ]
         else:
             catchment_water_quality_results = []
+
+    def catchment_intersects_aoi(aoi, geom):
+        # Clamp minimum intersection percentage for the catchment between 1 and
+        # 5%, with the percentage sliding from 1 up to 5 if the AOI's > 1000
+        # sq km in size (deriving from the AOI sq km value / 1000).
+        aoi_kms = aoi.area / 1000000
+        min_intersection_pct = 1 if aoi_kms < 1000 else min(5, aoi_kms / 1000)
+        catchment_geom = GEOSGeometry(json.dumps(geom), srid=4326)
+        catchment_geom.transform(5070, clone=True)
+
+        if not catchment_geom.valid:
+            return False
+
+        if not catchment_geom.intersects(aoi):
+            return False
+
+        intersection_pct = (catchment_geom.intersection(aoi).area /
+                            catchment_geom.intersection(aoi).area * 100)
+
+        return intersection_pct > min_intersection_pct
+
+    geom.transform(5070)
```

Comment:
Consider setting `clone` on this and referencing the result in a var that distinguishes it from the input.  Also consider transforming it immediately after creating it, so it's clear what its contents are.